worker-test/src/test/java/com/hpe/caf/worker/workertest/AdjustRetryLimitIT.java

```diff
@@ -0,0 +1,140 @@
+        try(final Connection connection = connectionFactory.newConnection()) {
+
+            final Channel channel = connection.createChannel();
+
+            channel.queueDeclare("testworker-out", true, false, false, Collections.emptyMap());
+
+            final TestWorkerQueueConsumer poisonConsumer = new TestWorkerQueueConsumer();
+            channel.basicConsume("testworker-out", true, poisonConsumer);
+
+            final Map<String, Object> belowRetryLimitHeaders = new HashMap<>();
+            belowRetryLimitHeaders.put("x-caf-worker-retry-limit", 10);
+            belowRetryLimitHeaders.put("x-caf-worker-retry", 2);
+
+            final Map<String, Object> aboveRetryLimitHeaders = new HashMap<>();
+            aboveRetryLimitHeaders.put("x-caf-worker-retry-limit", 2);
+            aboveRetryLimitHeaders.put("x-caf-worker-retry", 3);
+
+            final AMQP.BasicProperties belowRetryLimitProperties = new AMQP.BasicProperties.Builder()
+                    .headers(belowRetryLimitHeaders)
+                    .contentType("application/json")
+                    .deliveryMode(2)
+                    .priority(1)
+                    .build();
+
+            final AMQP.BasicProperties aboveRetryLimitProperties = new AMQP.BasicProperties.Builder()
+                    .headers(aboveRetryLimitHeaders)
+                    .contentType("application/json")
+                    .deliveryMode(2)
+                    .priority(1)
+                    .build();
+
+            final String body = "{\"version\":1,\"taskId\":\"1.1\",\"taskClassifier\":\"testWorkerIdentifier\",\"taskApiVersion\":1," +
+                    "\"taskData\":\"cG9pc29uIG1lc3NhZ2U\",\"taskStatus\":\"RESULT_SUCCESS\",\"context\":{},\"to\":\"worker-in\"," +
+                    "\"tracking\":null,\"sourceInfo\":{\"name\":\"worker-test\",\"version\":\"1.0.0\"}}";
+
+            channel.basicPublish("", "worker-in", belowRetryLimitProperties, body.getBytes(StandardCharsets.UTF_8));
+            channel.basicPublish("", "worker-in", aboveRetryLimitProperties, body.getBytes(StandardCharsets.UTF_8));
+
+            try {
+                for (int i=0; i<100; i++){
+
+                    Thread.sleep(100);
+
+                    if (poisonConsumer.getResponseTaskData().size() == 2){
+                        break;
+                    }
+                }
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+
+            Assert.assertNotNull(poisonConsumer.getResponseTaskData());
+
+            final ArrayList<String> messageResponses = poisonConsumer.getResponseTaskData();
+
+            Assert.assertTrue(messageResponses.get(0).contains(TEST_WORKER_RESULT));
+            Assert.assertTrue(messageResponses.get(1).contains(POISON_ERROR_MESSAGE));
+        }
+    }
+    private static class TestWorkerQueueConsumer implements Consumer {
```

Comment:
When you split the existing test into two you can reuse the pre-existing implementation of TestWorkerQueueConsumer